tests/MongoDB.Driver.Core.Tests/Core/Clusters/DnsClientWrapperTests.cs

```diff
@@ -171,7 +172,7 @@ public void ResolveTxtRecords_should_throw_when_cancellation_is_already_requeste
                 exception = Record.Exception(() => subject.ResolveTxtRecordsAsync(domainName, cts.Token).GetAwaiter().GetResult());
             }
 
-            exception.Should().BeOfType<OperationCanceledException>();
+            exception.Should().Match(c => c is OperationCanceledException || c is DnsResponseException);
```

Comment:
Now, dns client can return two exceptions. In the case of DnsResponseException, the OperationCanceledException will be in innerException. See for details: [link](https://evergreen.mongodb.com/task_log_raw/dot_net_driver_secure_tests__version~2.6_os~windows_64_topology~replicaset_auth~auth_ssl~ssl_test_patch_cafa20474a43e7a3dbac23416b2262f59c761f29_5e7e7d5a2fbabe2bbde0ed35_20_03_27_22_25_31/0?type=T#L1503) where I printed the thrown error in the EG log.